### PR TITLE
Tab-completion for panels and turbines

### PR DESCRIPTION
**Pull request by coroa**
Supersedes #39 ...

### CODE REVIEW

**Comment by coroa** [comment]
@FabianHofmann Had to continue here, since I was missing write access.

Please test and edit how you see fit!

**Comment by FabianHofmann** [comment]
Seems to be fine now! Using globals and an update function makes most sense

**Comment by coroa** [comment]
Ups, sorry, I *always* forget to fix imports.

**Comment by coroa** [comment]
@euronion I touched quite some of your code! Are you fine with the changes?

File: atlite/resource.py [hunk 1]
```diff
@@ -180,3 +184,172 @@ def smooth(velocities, power):
...
+
+        _update_resource_dictionaries()
+        logger.info(f"Turbine configuration downloaded to '{filepath}'.")
+
+
+    return turbineconf
+
+# Global caches
+_oedb_turbines = None
+turbines = arrowdict()
```
Comment (euronion) [inline atlite/resource.py]:
```suggestion
windturbines = arrowdict()
```
For consistency purposes.

File: atlite/resource.py [hunk 2]
```diff
@@ -180,3 +184,172 @@ def smooth(velocities, power):
...
+panels = arrowdict()
```
Comment (euronion) [inline atlite/resource.py]:
```suggestion
solarpanels = arrowdict()
```
For consistency purposes.

**Comment by euronion** [comment]
> @euronion I touched quite some of your code! Are you fine with the changes?

@coroa : Looks mostly good, thanks for that easy observer implementation!
The callback now is only activated when calling the `config.update()` method.
This would not activate if I were to directly assign values like `config.windturbine_dir = "/tmp"` .
Can this be fixed easily? Maybe extending on `__setattr__`?

File: atlite/resource.py (hunk 1 above)
Comment (coroa) [reply atlite/resource.py]: Food for thought on consistency:
```python
cutout.wind(turbine=atlite.windturbines.Vestas_V112_3MW)
cutout.pv(panel=atlite.solarpanels.CdTe)
```

**Comment by coroa** [comment]
>  Can this be fixed easily? Maybe extending on `__setattr__`?

I don't think `__setattr__` is called on module level (but feel free to try, i am unsure). We could package everything up into an object! :)

An easy solution would be to encourage updating config settings with f.ex.
```python
config.update(windturbine_dir="/tmp")
```
instead.

File: atlite/resource.py (hunk 1 above)
Comment (euronion) [reply atlite/resource.py]: `wind(turbine` and `pv(panel` sound nice IMHO.

Comment (coroa) [reply atlite/resource.py]: Sold!

**Pull request merged**